src/libsyntax/ext/tt/macro_rules.rs

```diff
@@ -790,6 +790,20 @@ fn is_in_follow(tok: &quoted::TokenTree, frag: &str) -> Result<bool, (String, &'
                 // harmless
                 Ok(true)
             },
+            "vis" => {
+                // Explicitly disallow `priv`, on the off chance it comes back.
+                match *tok {
+                    TokenTree::Token(_, ref tok) => match *tok {
+                        Comma => Ok(true),
+                        ModSep => Ok(true),
```

Comment:
If I remember correctly, I allowed `ModSep` because you can already have `pub ::path::to::a_type` that you might want to parse and special-case.  I allowed comma because having *some* kind of sequence-delimiting token is useful.  If you're passing bits of parsed input around, you might have `inner!($thing, $a_vis, $more_tts)`.  Everything else was excluded based on wanting to err on the side of being conservative.